### PR TITLE
Notifications with tests

### PR DESCRIPTION
**Pull request by boaz-codota**

### CODE REVIEW

**Review by dimacodota — APPROVED** [review]
LGTM

File: src/notifications/pollNotifications.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    return;
+  }
+
+  notifications.notifications.forEach(
+    (notification) => void handleNotification(notification)
+  );
+}
+
+async function handleNotification(notification: Notification): Promise<void> {
+  return vscode.window
```
Comment (dimacodota) [inline src/notifications/pollNotifications.ts:30]: let's add a `once` wrapper on top of it with a notification.id as a key

**Review by dimacodota — APPROVED** [review]
see my comment

File: src/notifications/pollNotifications.ts (hunk 1 above)
Comment (boaz-codota) [reply src/notifications/pollNotifications.ts:30]: Why? That sounds like a hack that will bite us in the ass in the future

**Pull request merged**